BUG- support pandas series

**Description**
Running mypy generates some errors: 
```
mypy --ignore-missing-imports --implicit-optional --no-namespace-packages --exclude '(taipy/templates/|generate_pyi.py|tools)' --follow-imports skip --disable-error-code import-untyped --show-column-numbers --hide-error-context .
```
Some errors relate to the returned type of the `_convert_data_to_dataframe` method in `_abstract_tabular`, `Union[pd.DataFrame, pd.Series]`.

The error is that the callers cannot accept a `pd.Series` type.

```
taipy\core\data\sql_table.py:126:13: error: 
Argument 1 to "__insert_dataframe" of "SQLTableDataNode" has incompatible type "DataFrame | Series[Any]"; expected "DataFrame"  [arg-type]
taipy\core\data\parquet.py:250:18: error: 
Incompatible types in assignment (expression has type "DataFrame | Series[Any]", variable has type "DataFrame")  [assignment]
```

**Acceptance Criteria**
- [ ] Exhibit the bug in a unit test
- [ ] Fix the issue

--- a/taipy/core/data/csv.py
+++ b/taipy/core/data/csv.py
@@ -188,7 +188,7 @@ class CSVDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
             )
         else:
             self._convert_data_to_dataframe(exposed_type, data).to_csv(
-                self._path, index=False, encoding=self.properties[self.__ENCODING_KEY], header=None
+                self._path, index=False, encoding=self.properties[self.__ENCODING_KEY], header=False
             )
 
     def write_with_column_names(self, data: Any, columns: Optional[List[str]] = None, job_id: Optional[JobId] = None):
@@ -201,6 +201,6 @@ class CSVDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
         """
         df = self._convert_data_to_dataframe(self.properties[self._EXPOSED_TYPE_PROPERTY], data)
         if columns and isinstance(df, pd.DataFrame):
-            df.columns = columns
+            df.columns = pd.Index(columns, dtype="object")
         df.to_csv(self._path, index=False, encoding=self.properties[self.__ENCODING_KEY])
         self.track_edit(timestamp=datetime.now(), job_id=job_id)

--- a/taipy/core/data/excel.py
+++ b/taipy/core/data/excel.py
@@ -301,7 +301,7 @@ class ExcelDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
                 if columns:
                     data[key].columns = columns
 
-                df.to_excel(writer, key, index=False, header=self.properties[self._HAS_HEADER_PROPERTY] or None)
+                df.to_excel(writer, key, index=False, header=self.properties[self._HAS_HEADER_PROPERTY] or False)
 
     def _write(self, data: Any):
         if isinstance(data, Dict):

--- a/taipy/core/data/parquet.py
+++ b/taipy/core/data/parquet.py
@@ -214,10 +214,10 @@ class ParquetDataNode(DataNode, _FileDataNodeMixin, _TabularDataNodeMixin):
         }
         kwargs.update(self.properties[self.__WRITE_KWARGS_PROPERTY])
         kwargs.update(write_kwargs)
-        if isinstance(data, pd.Series):
-            df = pd.DataFrame(data)
-        else:
-            df = self._convert_data_to_dataframe(self.properties[self._EXPOSED_TYPE_PROPERTY], data)
+
+        df = self._convert_data_to_dataframe(self.properties[self._EXPOSED_TYPE_PROPERTY], data)
+        if isinstance(df, pd.Series):
+            df = pd.DataFrame(df)
 
         # Ensure that the columns are strings, otherwise writing will fail with pandas 1.3.5
         df.columns = df.columns.astype(str)

--- a/taipy/core/data/sql_table.py
+++ b/taipy/core/data/sql_table.py
@@ -10,7 +10,7 @@
 # specific language governing permissions and limitations under the License.
 
 from datetime import datetime, timedelta
-from typing import Any, Dict, List, Optional, Set
+from typing import Any, Dict, List, Optional, Set, Union
 
 import pandas as pd
 from sqlalchemy import MetaData, Table
@@ -146,8 +146,14 @@ class SQLTableDataNode(_AbstractSQLDataNode):
         connection.execute(table.insert(), data)
 
     @classmethod
-    def __insert_dataframe(cls, df: pd.DataFrame, table: Any, connection: Any, delete_table: bool) -> None:
-        cls.__insert_dicts(df.to_dict(orient="records"), table, connection, delete_table)
+    def __insert_dataframe(
+        cls, df: Union[pd.DataFrame, pd.Series], table: Any, connection: Any, delete_table: bool
+    ) -> None:
+        if isinstance(df, pd.Series):
+            data = [df.to_dict()]
+        elif isinstance(df, pd.DataFrame):
+            data = df.to_dict(orient="records")
+        cls.__insert_dicts(data, table, connection, delete_table)
 
     @classmethod
     def __delete_all_rows(cls, table: Any, connection: Any, delete_table: bool) -> None:

